File: MINJU-KIMmm/src/PGS/dfs-bfs/타겟넘버.py
def dfs(numbers, target, index, sum):
    if len(numbers) == index:
        return 1 if sum == target else 0
    else:
        return dfs(numbers, target, index + 1, sum + numbers[index]) + dfs(numbers, target, index + 1,
                                                                           sum - numbers[index])


def solution(numbers, target):
    answer = 0
    answer = dfs(numbers, target, 0, 0)
    return answer

File: siyeonkm/src/PGS/DFS&BFS/타겟넘버.cpp
#include <string>
#include <vector>

using namespace std;

int cnt = 0;

void dfs(vector<int> &numbers, int target, int idx, int sum) {
    if(idx == numbers.size()) {
        if(sum == target) cnt++;
        return;
    }

    dfs(numbers, target, idx+1, sum+numbers[idx]);
    dfs(numbers, target, idx+1, sum-numbers[idx]);

    return;

}

int solution(vector<int> numbers, int target) {
    int answer = 0;

    dfs(numbers, target, 0, 0);
    answer = cnt;

    return answer;
}